pyppeteer/page.py

```diff
@@ -85,113 +66,115 @@ class Page(EventEmitter):
 
         if screenshotTaskQueue is None:
-            screenshotTaskQueue = list()
+            screenshotTaskQueue = []
         self._screenshotTaskQueue = screenshotTaskQueue
 
         self._workers: Dict[str, Worker] = dict()
+        self._disconnectPromise = None
 
         def _onTargetAttached(event: Dict) -> None:
             targetInfo = event['targetInfo']
             if targetInfo['type'] != 'worker':
                 # If we don't detach from service workers, they will never die.
                 try:
-                    client.send('Target.detachFromTarget', {
-                        'sessionId': event['sessionId'],
-                    })
+                    client.send(
+                        'Target.detachFromTarget', {
+                            'sessionId': event['sessionId'],
+                        })
                 except Exception as e:
                     debugError(logger, e)
                 return
             sessionId = event['sessionId']
-            session = client._createSession(targetInfo['type'], sessionId)
+            session = Connection.fromSession(client).session(sessionId)
             worker = Worker(
                 session,
                 targetInfo['url'],
                 self._addConsoleMessage,
                 self._handleException,
             )
             self._workers[sessionId] = worker
-            self.emit(Page.Events.WorkerCreated, worker)
+            self.emit(Events.Page.WorkerCreated, worker)
 
         def _onTargetDetached(event: Dict) -> None:
             sessionId = event['sessionId']
             worker = self._workers.get(sessionId)
             if worker is None:
                 return
-            self.emit(Page.Events.WorkerDestroyed, worker)
+            self.emit(Events.Page.WorkerDestroyed, worker)
             del self._workers[sessionId]
 
         client.on('Target.attachedToTarget', _onTargetAttached)
         client.on('Target.detachedFromTarget', _onTargetDetached)
 
         _fm = self._frameManager
-        _fm.on(FrameManager.Events.FrameAttached,
-               lambda event: self.emit(Page.Events.FrameAttached, event))
-        _fm.on(FrameManager.Events.FrameDetached,
-               lambda event: self.emit(Page.Events.FrameDetached, event))
-        _fm.on(FrameManager.Events.FrameNavigated,
-               lambda event: self.emit(Page.Events.FrameNavigated, event))
-
-        _nm = self._networkManager
+        _fm.on(Events.FrameManager.FrameAttached,
+               lambda event: self.emit(Events.Page.FrameAttached, event))
+        _fm.on(Events.FrameManager.FrameDetached,
+               lambda event: self.emit(Events.Page.FrameDetached, event))
+        _fm.on(Events.FrameManager.FrameNavigated,
+               lambda event: self.emit(Events.Page.FrameNavigated, event))
+
+        networkManager = self._frameManager.networkManager
+        _nm = networkManager
         _nm.on(NetworkManager.Events.Request,
```

Comment:
Shouldn't this line be this? Same goes for the next 8 event listeners.
```suggestion
        _nm.on(Events.NetworkManager.Request,
```